witness/src/Dscp/Witness/Logic/Getters.hs

```diff
@@ -36,21 +37,33 @@ import Dscp.Witness.Launcher.Context
 import Dscp.Witness.Logic.Exceptions
 import Dscp.Witness.Mempool ()
 import Dscp.Witness.SDLock
+import Dscp.Witness.Web.Types (BlocksOrMempool (..))
+
+-- | Run an action both in chain-only and chain+mempool modes.
+runSdDual
+    :: WitnessWorkMode ctx m
+    => (forall mode. KnownSdReadMode mode => SdReadM mode m a)
+    -> m (BlocksOrMempool a)
+runSdDual action =
+    sequence BlocksOrMempool
+    { bmConfirmed = runSdReadM @'ChainOnly action
+    , bmTotal = runSdReadM @'ChainAndMempool action
+    }
 
 ----------------------------------------------------------------------------
 -- Block/Header/Tip getters
 ----------------------------------------------------------------------------
 
 -- | Retrieves current tip.
-getTipHash :: HasWitnessConfig => SdM_ chgacc HeaderHash
+getTipHash :: HasWitnessConfig => SdM HeaderHash
```

Comment:
How much sense does it make to restrict all those methods only to one possible `chgacc`?

Reply:
It seems to me that now when we support both chain-only and chain+mempool modes, requirements to `chgacc` should not change anymore.
They were actually restricted to one `chgacc` yet just a couple of weeks ago (before mempool PR), if anyone ever needs polymorphic `chgacc` back, it's quick to return.